rs/types/types/src/consensus/idkg.rs

```diff
@@ -92,7 +170,7 @@ pub struct IDkgPayload {
     pub xnet_reshare_agreements: BTreeMap<IDkgReshareRequest, CompletedReshareRequest>,
 
     /// State of the key transcripts.
-    pub key_transcripts: BTreeMap<MasterPublicKeyId, MasterKeyTranscript>,
+    pub key_transcripts: BTreeMap<IDkgMasterPublicKeyId, MasterKeyTranscript>,
```

Comment:
How come you can change the type of these structs here and below, without the hash function changing (and therefore the `cup_compatibility_test` failing)?

Reply:
Not 100% sure, but Hash uses serialization, right? The serialization of the `IDkgMasterpublicKeyId` is a custom implementation to make it transparent, i.e. it will look serialized exactly like a `MasterPublicKeyId`.

Reply:
Ah yes, that's probably it